feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,562 @@
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+	eID := gribi.BecomeLeader(t, client)
+
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(client); err != nil {
+		t.Fatal(err)
+	}
+
+	args := &testArgs{
+		ctx:        ctx,
+		client:     client,
+		dut:        dut,
+		ate:        ate,
+		top:        top,
+		electionID: eID,
+	}
+	// virtualVIPs are ipv4 entries in default vrf
+	pushDefaultEntries(t, args, subIntfIPs)
+
+	// Send traffic from ATE port-1 to prefixes in IPBlock1 and ensure traffic
+	// flows 100% and reaches ATE port-2.
+	flow := createTrafficFlow(t, args.ate, args.top)
+	testTraffic(t, *args, flow)
+
+	controllers := cmp.FindComponentsByType(t, dut, controlcardType)
+	t.Logf("Found controller list: %v", controllers)
+	// Only perform the switchover for the chassis with dual controllers.
+	if len(controllers) != 2 {
+		t.Skipf("Dual controllers required on %v: got %v, want 2", dut.Model(), len(controllers))
+	}
+
+	secondaryBeforeSwitch, primaryBeforeSwitch := findSecondaryController(t, dut, controllers)
+
+	if ok := switchoverReady(t, dut, primaryBeforeSwitch); !ok {
+		t.Fatalf("Controller %q did not become switchover-ready before test.", primaryBeforeSwitch)
+	}
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: cmp.GetSubcomponentPath(secondaryBeforeSwitch),
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+
+	entriesBefore := checkNIHasNEntries(ctx, client, *deviations.DefaultNetworkInstance, t)
+
+	// Concurrently run switchover and gribi route flush
+	t.Log("Execute gRIBi flush and master switchover concurrently")
+	go func(msg string) {
+		if _, err := gribi.Flush(client, eID, *deviations.DefaultNetworkInstance); err != nil {
+			t.Logf("Unexpected error from flush, got: %v", err)
+		}
+	}("gRIBi Flush")
+
+	go func(msg string) {
+		switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+		if err != nil {
+			t.Logf("Failed to perform control processor switchover with unexpected err: %v", err)
+		}
+		t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+	}("Master Switchover")
+
+	t.Log("Switchover/Flush is completed, validate switchoverStatus now")
+
+	primaryAfterSwitch := validateSwitchoverStatus(t, dut, secondaryBeforeSwitch)
+
+	validateSwitchoverTelemetry(t, dut, primaryAfterSwitch)
+
+	// Following reconnection of the gRIBI client to a new master supervisor,
+	// validate if partially deleted entries of IPBlock1  are not present in the FIB
+	// using a get RPC.
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	// Check vars for WithInitialElectionID
+
+	t.Log("Added wait time to make sure things are stablized post switchover")
```

Comment:
I do think this sleep is really needed since the above code validate the swictover status and ensure the new master is up.

Reply:
ok. sure

Reply:
Removed sleep. thanks